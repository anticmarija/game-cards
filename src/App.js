import React from 'react';
import { connect } from 'react-redux';

import {
  getNumberOfPlayers,
  getLoadingState,
} from './store/selectors/selectors';
import './App.css';
import GameIntro from './GameIntro/GameIntro';
import Player from './Player/Player';
import Board from './Board/Board';
import ComputerPlayer from './ComputerPlayer/ComputerPlayer';
import loadingGif from './assets/loading.gif';

export const App = ({ numberOfPlayers, loading }) => {
  return (
    <div className="game">
      {numberOfPlayers ? (
        loading ? (
          <img src={loadingGif} alt="loading" />
        ) : (
          <div className="game-board">
            <Board />
            <Player />
            {[...Array(numberOfPlayers - 1)].map((e, i) => (
              <ComputerPlayer key={i} player={`player${i + 2}`} />
            ))}
          </div>
        )
      ) : (
        <GameIntro />
      )}
    </div>
  );
};

export const mapStateToProps = state => ({
  numberOfPlayers: getNumberOfPlayers(state),
  loading: getLoadingState(state),
  state: state,
});

export default connect(mapStateToProps)(App);
