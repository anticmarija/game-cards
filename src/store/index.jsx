import { applyMiddleware, createStore, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import gameReducer from './reducers/game.reducer';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

const store = createStore(
  combineReducers({ game: gameReducer }),
  composeWithDevTools(applyMiddleware(thunk)),
);

export default store;
