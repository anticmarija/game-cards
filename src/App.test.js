import React from 'react';
import { render } from '@testing-library/react';
import { App, mapStateToProps } from './App';
import { Provider } from 'react-redux';
import store from './store';

describe('Test App component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <App />
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <App numberOfPlayers={2} />
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});

describe('mapStateToProps', () => {
  it('should provide number of players prop', () => {
    const state = { game: { numberOfPlayers: 2 } };

    expect(mapStateToProps(state)).toHaveProperty('numberOfPlayers');
  });
});
