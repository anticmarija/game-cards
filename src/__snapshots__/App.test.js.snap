// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component should render correctly 1`] = `
<div
  class="game"
>
  <h1>
    Select number of players
  </h1>
  <button
    class="button"
  >
    2
     players
  </button>
  <button
    class="button"
  >
    3
     players
  </button>
  <button
    class="button"
  >
    4
     players
  </button>
</div>
`;
